File: 31-ingress-spring/app/src/main/java/br/com/edu/socket/test/controller/TestController.java
/* Spring app (C)2023 */
package br.com.edu.socket.test.controller;

import br.com.edu.socket.test.controller.dto.ResponseDTO;
import java.io.IOException;
import java.net.*;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.server.ResponseStatusException;

@Slf4j
@RestController("/")
public class TestController {

  @GetMapping(value = "/ok", produces = "application/json")
  public ResponseEntity<ResponseDTO> ok() {
    return ResponseEntity.ok().body(new ResponseDTO(System.getenv("SERVER_NAME")));
  }

  @GetMapping(value = "/test", produces = "application/json")
  public ResponseEntity<ResponseDTO> test(
      @RequestParam(required = true) String host, @RequestParam(required = true) Integer port)
      throws IOException {

    try {
      log.debug("Iniciando o teste: " + host + ":" + port);
      HttpClient httpClient = HttpClient.newHttpClient();
      HttpRequest httpRequest =
          HttpRequest.newBuilder()
              .uri(URI.create("http://" + host + ":" + port + "/ok"))
              .GET()
              .build();

      HttpResponse<String> httpResponse =
          httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());

      String responseBody = httpResponse.body();

      return ResponseEntity.ok().body(new ResponseDTO(System.getenv("SERVER_NAME")));
    } catch (Exception e) {
      e.printStackTrace();
      log.error(e.getMessage());
      throw new ResponseStatusException(HttpStatus.NOT_FOUND, e.getMessage(), e);
    }
  }
}

File: java-tunning/src/main/kotlin/api/TestController.kt
package br.com.edu.api

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import java.lang.Math.pow
import kotlin.math.*

@RequestMapping("/v1")
@RestController
class TestController {

    @GetMapping("/test")
    fun test(@RequestParam(value = "value", required = false, defaultValue = "3.5") value: Double) : ResponseEntity<String> {

        println("Iniciando calculo")
        var result: Double = calculate(value)

        println("fim calculo")
        return ResponseEntity.ok("Value: $result")
    }

    fun calculate(value: Double): Double {
        var result: Double = 1.0
        for (i in 1..10_000_000) {
            result *= 1.123456.pow(sin(value) * cos(value) / tan(value)) / (atan(value))
        }
        return result * atan(cos(sin(value))) * 3 * PI
    }

}
